### PR TITLE
Implement DataView.prototype.{set, get}Big{U}int64 methods

### PR DESCRIPTION
**Pull request by szilagyiadam**
The algorithms are based on:
- ECMA-262 v11, 24.3.4.5
- ECMA-262 v11, 24.3.4.6
- ECMA-262 v11, 24.3.4.15
- ECMA-262 v11, 24.3.4.16

JerryScript-DCO-1.0-Signed-off-by: Adam Szilagyi aszilagy@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-dataview-prototype.inc.h [hunk 1]
```diff
@@ -56,6 +56,12 @@ ROUTINE (LIT_MAGIC_STRING_SET_INT32_UL, ECMA_DATAVIEW_PROTOTYPE_SET_INT32, 2, 1)
 ROUTINE (LIT_MAGIC_STRING_SET_UINT8_UL, ECMA_DATAVIEW_PROTOTYPE_SET_UINT8, 2, 1)
 ROUTINE (LIT_MAGIC_STRING_SET_UINT16_UL, ECMA_DATAVIEW_PROTOTYPE_SET_UINT16, 2, 1)
 ROUTINE (LIT_MAGIC_STRING_SET_UINT32_UL, ECMA_DATAVIEW_PROTOTYPE_SET_UINT32, 2, 1)
+#if ENABLED (JERRY_BUILTIN_BIGINT)
+ROUTINE (LIT_MAGIC_STRING_GET_BIGINT64, ECMA_DATAVIEW_PROTOTYPE_GET_BIGINT64, 2, 1)
+ROUTINE (LIT_MAGIC_STRING_GET_BIGUINT64, ECMA_DATAVIEW_PROTOTYPE_GET_BIGUINT64, 2, 1)
+ROUTINE (LIT_MAGIC_STRING_SET_BIGINT64, ECMA_DATAVIEW_PROTOTYPE_SET_BIGINT64, 2, 1)
+ROUTINE (LIT_MAGIC_STRING_SET_BIGUINT64, ECMA_DATAVIEW_PROTOTYPE_SET_BIGUINT64, 2, 1)
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-dataview-prototype.inc.h:63]: The length property should be `2` for setters. Please update the other methods as well.

File: tests/test262-esnext-excludelist.xml [hunk 2]
```diff
@@ -609,23 +583,12 @@
   <test id="built-ins/DataView/prototype/setBigInt64/detached-buffer-after-toindex-byteoffset.js"><reason></reason></test>
   <test id="built-ins/DataView/prototype/setBigInt64/detached-buffer-before-outofrange-byteoffset.js"><reason></reason></test>
   <test id="built-ins/DataView/prototype/setBigInt64/detached-buffer.js"><reason></reason></test>
-  <test id="built-ins/DataView/prototype/setBigInt64/index-check-before-value-conversion.js"><reason></reason></test>
-  <test id="built-ins/DataView/prototype/setBigInt64/index-is-out-of-range.js"><reason></reason></test>
-  <test id="built-ins/DataView/prototype/setBigInt64/length.js"><reason></reason></test>
-  <test id="built-ins/DataView/prototype/setBigInt64/name.js"><reason></reason></test>
-  <test id="built-ins/DataView/prototype/setBigInt64/negative-byteoffset-throws.js"><reason></reason></test>
   <test id="built-ins/DataView/prototype/setBigInt64/range-check-after-value-conversion.js"><reason></reason></test>
   <test id="built-ins/DataView/prototype/setBigInt64/return-abrupt-from-tobigint-value.js"><reason></reason></test>
```
Comment (rerobika) [inline tests/test262-esnext-excludelist.xml:587]: Can you please check these 2 failing testcases?

File: jerry-core/ecma/operations/ecma-dataview-object.c [hunk 3]
```diff
@@ -314,7 +314,21 @@ ecma_op_dataview_get_set_view_value (ecma_value_t view, /**< the operation's 'vi
     return ecma_get_typedarray_element (swap_block_p, id);
   }
 
-  if (ecma_is_value_number (value_to_set))
+#if ENABLED (JERRY_BUILTIN_BIGINT)
+  bool set_value_is_bigint = ecma_is_value_bigint (value_to_set);
+
+  if (id > ECMA_FLOAT64_ARRAY)
+  {
+    if (!set_value_is_bigint)
```
Comment (dbatyai) [inline jerry-core/ecma/operations/ecma-dataview-object.c:322]: The value should be coerced to a bigint, not just checked. In fact, it seems this whole method needs an update, as regular values should also be coerced to a number, and the coercion should happen much earlier.

**Review by dbatyai — CHANGES_REQUESTED** [review]

**Comment by ossy-szeged** [comment]
It is outdated and will be fixed in #4287 and #4315.

**Pull request closed without merging**